12-monads/Risk.hs

```diff
@@ -0,0 +1,172 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Risk where
+
+import Control.Monad.Random
+import Data.List
+
+------------------------------------------------------------
+-- Die values
+
+newtype DieValue = DV { unDV :: Int } 
+  deriving (Eq, Ord, Show, Num)
+
+first :: (a -> b) -> (a, c) -> (b, c)
```

Comment:
@jhrcek Thanks, yeap not really sure why the question provided its implementation as well.

Reply:
I think it's exactly for this reason: https://www.michaelpj.com/blog/2021/01/02/elementary-programming.html
It's tradeoff between simple code for clarity (not introducing new abstractions because students doing this homework already have ton of new stuff they're learning) vs. using abstraction to avoid reimplementing new concepts (but that requires learning and understanding the abstraction)

Reply:
@jhrcek This figures. How can we doubt the great Brent Yorgey!